### PR TITLE
`Push, pull, compact, verify` commands e2e tests

### PR DESCRIPTION
**Pull request by rmn-boiko**
## Description

Closes: #issue

## Checklist before requesting a review

- [x] Unit and integration tests added
  
- [x] Compatibility:
    
  - [x] Network APIs: ✅
    
  - [x] Workspace layout and metadata: ✅
    
  - [x] Configuration: ✅
    
  - [x] Container images: ✅
- [x] Observability:
    
  - [x] Tracing / [metrics](https://github.com/kamu-data/kamu-standards/blob/master/metrics_design.md): ✅
    
  - [x] Alerts: ✅
- [x] Documentation:
    
  - [x] [Changelog](./CHANGELOG.md): ✅
    
  - [x] [Public documentation](https://github.com/kamu-data/kamu-docs/): ✅
  
- [x] Downstream effects:
    
  - [x] [kamu-node](https://github.com/kamu-data/kamu-node): ✅
    
  - [x] [kamu-web-ui](https://github.com/kamu-data/kamu-web-ui): ✅
    
  - [x] [release-queue](https://github.com/orgs/kamu-data/projects/4/views/1)

### CODE REVIEW

File: src/e2e/app/cli/repo-tests/src/test_smart_transfer_protocol.rs
Comment (s373r) [inline src/e2e/app/cli/repo-tests/src/test_smart_transfer_protocol.rs:1]: Please add also a test for pulling the deriv datasets smth like:
image

**Review by s373r — COMMENTED** [review]
Added minor comments

File: src/e2e/app/cli/repo-tests/src/test_smart_transfer_protocol.rs [hunk 1]
```diff
@@ -140,3 +105,1100 @@ pub async fn test_smart_push_pull_sequence(kamu_api_server_client: KamuApiServer
...
+        let mut dataset_endpoint = Url::parse("odf+http://host").unwrap();
+
+        dataset_endpoint.set_host(base_url.host_str()).unwrap();
+        dataset_endpoint.set_port(base_url.port()).unwrap();
+
+        dataset_endpoint
+            .join(format!("e2e-user/{dataset_name}").as_str())
+            .unwrap()
+            .to_string()
+    };
```
[the remark this answers is not in the document]
Comment (rmn-boiko) [reply src/e2e/app/cli/repo-tests/src/test_smart_transfer_protocol.rs:126]: Changed

File: src/e2e/app/cli/common/src/kamu_api_server_client_ext.rs [hunk 2]
```diff
@@ -249,14 +254,21 @@ impl KamuApiServerClientExt for KamuApiServerClient {
...
+        &self,
+        token: &AccessToken,
+        account_name_maybe: Option<AccountName>,
+    ) -> DatasetId {
         let dataset_id = self.create_player_scores_dataset(token).await;
 
         // TODO: Use the alias from the reply, after fixing the bug:
         //       https://github.com/kamu-data/kamu-cli/issues/891
 
         // At the moment, only single-tenant
```
Comment (s373r) [inline src/e2e/app/cli/common/src/kamu_api_server_client_ext.rs:267]: > // At the moment, only single-tenan

it doesn't seem to be any more relevant

File: src/e2e/app/cli/repo-tests/src/test_smart_transfer_protocol.rs [hunk 3]
```diff
@@ -1148,57 +1131,185 @@ pub async fn test_smart_push_pull_s3(mut kamu: KamuCliPuppet) {
...
         );
         kamu.assert_last_data_slice(&dataset_name, expected_schema, expected_data)
             .await;
     }
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub async fn test_smart_pull_derivative(mut kamu: KamuCliPuppet) {
+    kamu.set_system_time(Some(DateTime::from_str("2000-01-01T00:00:00Z").unwrap()));
```
Comment (s373r) [inline src/e2e/app/cli/repo-tests/src/test_smart_transfer_protocol.rs:1192]: Optional: can be set via test options (in the macro)

File: src/utils/kamu-cli-puppet/src/kamu_cli_puppet_ext.rs [hunk 4]
```diff
@@ -244,8 +253,7 @@ pub struct DatasetRecord {
     #[serde(rename = "ID")]
     pub id: DatasetID,
     pub name: DatasetName,
-    #[serde(with = "DatasetKindDef")]
-    pub kind: DatasetKind,
+    pub kind: String,
```
Comment (s373r) [inline src/utils/kamu-cli-puppet/src/kamu_cli_puppet_ext.rs:256]: Q: why is it necessary?

**Review by s373r — COMMENTED** [review]
Minor comments & a question added

**Review by s373r — APPROVED** [review]

**Pull request merged**